addon/mixins/object-browser-mixin.js

```diff
@@ -0,0 +1,201 @@
+          facetsConfig: facetsConfig,
+          selectedItemsCount: this.get('selectedItemsCount'),
+          onFilterFormChange: this.get('_onFilterFormChange')
+        }
+      }))
+  }),
+
+  // cp: count for currently selected items
+  selectedItemsCount: Ember.computed('selectedItems', function () {
+    return Object.keys(this.get('selectedItems')).length
+  }),
+
+  // remove list selection state by setting selectedItems to Ember.A()
+  clearListState: function () {
+    if (this.get('selectedItems')) {
+      this.set('selectedItems', Ember.A())
+    }
+  },
+
+  // default filter method for object browser
+  objectBrowserDefaultFilter: function (data, filter) {
+    if (!isPresent(filter)) {
+      return data
+    }
+    return data.filter((data) => {
+      let qualified = true
+      let keys = Object.keys(filter)
+
+      keys.forEach(key => {
+        if (data.get(key).indexOf(filter[key]) === -1) {
+          qualified = false
+        }
+      })
+
+      return qualified
+    })
+  },
+
+  // default sort method for object browser
+  objectBrowserDefaultSort: function (items, sortProperties) {
+    if (!isPresent(sortProperties)) {
+      return
+    }
+
+    let normalizedSortProperties = sortProperties.map(sortProperty => {
+      let resultArray = []
+      if (sortProperty.startsWith('-')) {
+        resultArray.pushObject(sortProperty.slice(1))
+        resultArray.pushObject('desc')
+      } else {
+        resultArray.pushObject(sortProperty)
+        resultArray.pushObject('asc')
+      }
+      return resultArray
+    })
+
+    return Ember.A(items.slice().sort((itemA, itemB) => {
+      for (let i = 0; i < normalizedSortProperties.length; i++) {
+        let [prop, direction] = normalizedSortProperties[i]
+        let result = Ember.compare(Ember.get(itemA, prop), Ember.get(itemB, prop))
+        if (result !== 0) {
+          return (direction === 'desc') ? (-1 * result) : result
+        }
+      }
+      return 0
+    }))
+  },
+
+  actions: {
+    sortItems (sortItems) {
+      // create serializer
+      const serializer = this.get('objectBrowserConfig.serializerConfig.serializer').create({
+        config: this.get('objectBrowserConfig.serializerConfig'),
+        context: this
+      })
+
+      // TODO This is used to get rid of extra properties coming from ember object. Remove this when sort component improved.
+      let activeSorting = sortItems.map(function (item) {
+        return {value: item.value, direction: item.direction}
+      })
+
+      // set state prop which will feed back to component
+      // this.set('activeSorting', activeSorting)
+
+      const normalizedSorting = serializer.normalizeSort(activeSorting)
+
+      this.setProperties({
+        sortQueryParam: normalizedSorting,
+        pageQueryParam: []
+      })
+    },
+
+    filterItems (formValue) {
+      this.set('objectBrowserConfig.facetsConfig.value', formValue)
+      // create serializer
```

Comment:
Should we isolate this in a method since we are copying the same 3 lines above 144-147?